jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/bigquery/BigQuerySqlDialect.java

```diff
@@ -0,0 +1,136 @@
+                        LEAST, LN, LOG, MOD, POWER, RAND, ROUND, SIGN, SIN, SINH, SQRT, TAN, TANH, TRUNC,
+                        COLOGNE_PHONETIC, CONCAT, INSERT, INSTR, LENGTH, LOWER, LPAD, LTRIM, REGEXP_REPLACE, REPEAT,
+                        REPLACE, REVERSE, RIGHT, RPAD, RTRIM, SOUNDEX, SPACE, SUBSTR, TRIM, UPPER, CURRENT_DATE,
+                        CURRENT_TIMESTAMP, DATE_TRUNC, DAY, EXTRACT, MINUTE, MONTH, SECOND, WEEK, YEAR, ST_X, ST_Y,
+                        ST_LENGTH, ST_NUMPOINTS, ST_AREA, ST_BOUNDARY, ST_CENTROID, ST_CONTAINS, ST_DIFFERENCE,
+                        ST_DIMENSION, ST_DISJOINT, ST_DISTANCE, ST_EQUALS, ST_INTERSECTION, ST_INTERSECTS, ST_ISEMPTY,
+                        ST_TOUCHES, ST_UNION, ST_WITHIN, CAST, TO_TIMESTAMP, BIT_AND, BIT_OR, BIT_XOR, CASE, HASH_MD5) // TODO
+                                                                                                                       // md5
+                                                                                                                       // translation
+                .addAggregateFunction(COUNT, COUNT_STAR, COUNT_DISTINCT, SUM, SUM_DISTINCT, MIN, MAX, AVG, AVG_DISTINCT,
+                        FIRST_VALUE, LAST_VALUE, STDDEV, STDDEV_DISTINCT, STDDEV_POP, STDDEV_POP_DISTINCT, STDDEV_SAMP,
+                        STDDEV_SAMP_DISTINCT, VARIANCE, VARIANCE_DISTINCT, VAR_POP, VAR_POP_DISTINCT, VAR_SAMP,
+                        VAR_SAMP_DISTINCT, GROUP_CONCAT, APPROXIMATE_COUNT_DISTINCT) // TODO
+                                                                                     // APPROXIMATE_COUNT_DISTINCT
+                                                                                     // translation
+                .build();
+    }
+
+    @Override
+    protected List<String> getSupportedProperties() {
+        return SUPPORTED_PROPERTIES;
+    }
+
+    @Override
+    public Capabilities getCapabilities() {
+        return CAPABILITIES;
+    }
+
+    @Override
+    public StructureElementSupport supportsJdbcCatalogs() {
+        return StructureElementSupport.MULTIPLE;
+    }
+
+    @Override
+    public StructureElementSupport supportsJdbcSchemas() {
+        return StructureElementSupport.MULTIPLE;
+    }
+
+    @Override
+    public String applyQuote(String identifier) {
+        return "`" + identifier + "`";
+    }
+
+    @Override
+    public boolean requiresCatalogQualifiedTableNames(SqlGenerationContext context) {
+        return true;
+    }
+
+    @Override
+    public boolean requiresSchemaQualifiedTableNames(SqlGenerationContext context) {
+        return true;
+    }
+
+    @Override
+    public NullSorting getDefaultNullSorting() {
+        return NullSorting.NULLS_SORTED_AT_END;
+    }
+
+    @Override
+    public String getStringLiteral(String value) {
+        final StringBuilder builder = new StringBuilder("'");
```

Comment:
This looks like code duplication with the default implementation. Pull up to `AbstractSqlDialect` and remove override in all dialect that do it the same way.

Reply:
done